packages/editor-ui/src/components/MainHeader/MainHeader.vue

```diff
@@ -124,11 +121,15 @@ export default defineComponent({
 				this.activeHeaderTab = MAIN_HEADER_TABS.WORKFLOW;
 			}
 
-			if (to.params.name !== 'new') {
+			if (to.params.name !== 'new' && typeof to.params.name === 'string') {
 				this.workflowToReturnTo = to.params.name;
 			}
 
-			if (from?.name === VIEWS.EXECUTION_PREVIEW && to.params.name === from.params.name) {
+			if (
+				from?.name === VIEWS.EXECUTION_PREVIEW &&
+				to.params.name === from.params.name &&
+				typeof from.params.executionId === 'string'
```

Comment:
is there any chance this can be a number? since execution ids are usually numbers?

Reply:
Not really. It's a router param. It's a string and could be an array of strings, so I had to put that condition there.

Or I can change `executionToReturnTo` in this component to be of type `RouteLocation` instead of `string`, but then I have to change `string` everywhere where this component uses a route param.